src/core/packages/metrics/server-internal/src/exponential_moving_average.ts

```diff
@@ -25,11 +29,14 @@ export function exponentialMovingAverage(
 
   return (inner) => {
     let previous: number | undefined;
+    let mean = 0;
 
     return inner.pipe(
-      map((current) => (previous == null ? current : alpha * current + (1 - alpha) * previous)),
-      tap((current) => {
-        previous = current;
+      map((current, index) => {
+        if (index < period / interval) {
+          return (mean += (current * interval) / period); // accumulating mean value
```

Comment:
Would the acc. mean not be calculated by:

```ts
mean += (current - mean) / index
```

Or if we want to weight in the "0" observations for an even smaller nr

```ts
mean += (current - mean) / (period /* e.g. 15s */ / interval /* e.g. 5s */)
```

...otherwise the current calculation will be an ever-increasing number. For e.g. I would expect this mean behaviour

```
1st collection
current: 2
mean: 2

2nd collection (lower)
current: 1
mean: 1.5

3rd collection (higher)
current: 2
mean: 1.67
```


Not a stats expert though and may have misundesrtood! 😄

Reply:
Likely this is more a question for @dokmic